docs/topics/proc_exposing-application-metrics-using-prometheus-on-node-js.adoc

```diff
@@ -0,0 +1,132 @@
+
+[id='exposing-application-metrics-using-prometheus-on-node-js']
+= Exposing application metrics using Prometheus on {Node}
+
+In this example, you:
+
+* Configure your application to expose metrics.
+* Collect and view the data using Prometheus.
+
+Note that Prometheus actively connects to a monitored application to collect data; the application does not actively send metrics to a server.
+
+.Prerequisites
+
+* Prometheus configured to collect metrics from the application:
++
+--
+. Get the default Prometheus configuration using the following command:
++
+[source,bash,opts="nowrap"]
+----
+# docker run -it --rm --entrypoint /bin/sh prom/prometheus -c 'cat /etc/prometheus/prometheus.yml'
+----
++
+Store the output in the `prometheus.yml` file in the directory with your application.
+
+. Append the following snippet to the `prometheus.yml` file to make Prometheus automatically collect metrics from your application:
++
+[source,yaml]
+----
+  - job_name: nodejs
+     static_configs:
+     - targets: ['localhost:8888']
+----
++
+The default behavior of {Node}-based applications is to expose metrics at the `/metrics` endpoint.
+This is also what Prometheus expects.
+--
+
+* The Prometheus server running on `localhost` using Docker:
++
+--
+[source,bash,opts="nowrap"]
+----
+# docker run -it --rm --network host -p 9090:9090 -v $PWD/prometheus.yml:/etc/prometheus/prometheus.yml prom/prometheus
+----
+
+Wait until the `Server is ready to receive web requests` message is displayed in the console.
+
+Include the the `--network host` option to allow Prometheus to connect to your application.
+--
+
+.Procedure
+
+. Install the `prom_client` npm package.
++
+.pom.xml
+[source,bash]
+----
+$ npm install prom-client express body-parser --save
+----
+
+. Create an example {NodeJS} application and save it in a file with the file name `app.js` .
++
+--
+[source,javascript]
+----
+'use strict';
+const path = require('path');
+const express = require('express');
+const bodyParser = require('body-parser');
+const client = require('prom-client');
+
+// enable prom-client to expose default application metrics
+const collectDefaultMetrics = client.collectDefaultMetrics;
+
+// define a custom prefix string for application metrics
+collectDefaultMetrics({ prefix: 'application:' });
+
+const histogram = new client.Histogram({
+  name: 'application:hello',
```

Comment:
It's really a nitpicky comment, I know, but here I would also use underscore. `application_hello`. As an aside, I might choose a different, more descriptive prefix :)